### DISCUSSION TITLE
Associate habits with tags

### DISCUSSION THREAD

**Discussion by AnderRasoVazquez** [iSoron/uhabits#31] [Feature Requests]
Tags are great for organizing items, would be great to assign a tag to habits instead of colors. For example, all "health" related habits would have the same color that we assigned to health tag.

**Comment by iSoron** [comment]
I think tags are an interesting feature. It would allow us to implement, for example, combined statistics for groups of habits. But I would like to keep the colors independent from them, since you can usually assign multiple tags to one item.

**Reply by Kimmueses** [reply]
I think this is a great idea. I just downloaded the app and I already like it better than the one I paid for in the past. However, I do love the idea of having a combined statistics view of all or certain groups of habits like the old app I had. I have so many habits and don't expect to do all of them perfectly as life is unpredictable but if I have a combined average of at least 80% that would be amazing. The app I had was called Habit Tracker by Davetech Co on iOS in case you want to see how they did it.

**Comment by aleksejrs** [comment]
The worst thing about tags for me in some tagging UIs (apparently OI Shopping List suffers from this) is the inability to find items not tagged with any tag of a group or at all when I want to add them.  Please make it possible to find at least completely untagged items.

Assigning multiple tags would not worsen this much if it is possible to search for items not tagged with a specific tag, if Loop keeps the habit order consistent in all lists.

**Comment by aleksejrs** [comment]
> to find items not tagged with any tag of a group or at all when I want to add them.

Tickmate actually has a nice tagging UI, which includes both a list of tags to choose from for a track, and a list of tracks to choose from for a tag.

**Comment by mr-flibble** [comment]
Hello, I think the app would benefit from the ability to add tags to the items and be able to view only the selected tags or all (like in habit rpg android app https://play.google.com/store/apps/details?id=com.habitrpg.android.habitica&hl=cs ) 

Example: If I have a habit "go sleep early", so I never actually see a screen congratulating me that I have fulfilled everything.
While 90% of my day I do not care about this task.

**Comment by VA-MMXXII** [comment]
I 100% agree that some sort of tagging, labeling, categorization, or grouping would be very useful, but not so much for searching (as most users are unlikely to hundreds of habits).

Instead, I think grouping would be useful for pulling more stats, gamification, and for organizing/decluttering/focusing the UI with collapsible groups that could also have their own checkmark grid widgets.

#51 #1193

**Comment by toupeira** [comment]
Regarding the navigation UI for tags, this could be implemented like the navigation drawer in GMail where you can select labels and other folders. You could have filters like "All habits", "Pending habits", "Completed habits", then an entry for each tag, and then all the other items that are currently in the top-right menu.

The drawer could be opened through a hamburger menu like now, or maybe through a new bottom navigation bar.

Not sure if it's necessary to display the tags in the main list view, this could also be optional behind a toggle like Dark Mode.

**Comment by AnsahMohammad** [comment]
Can we also have the feature of grouping it as a folder . All the habits belonging to a tag can be grouped.

**Comment by lkraav** [comment]
Keep finding myself looking for this every time I use the app, probably the same with everyone who has spent a year or longer with it.

Un-grouped it all becomes a mess, and motivation to maintain checking in drops.

**Reply by SKIERZZ** [reply]
Ten years passed from this suggestion but the feature has not been added...
